### PR TITLE
Extend Contribution Guide

### PR DESCRIPTION
**Pull request by lukaszstolarczuk**
We should consider this as the first version of this document. It should be updated with more content as we go.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.md
Comment (igchor) [inline .github/ISSUE_TEMPLATE/bug_report.md:18]: Perhaps a 'git tag' for now since we don't really have any versions released yet?

File: CONTRIBUTING.md
Comment (igchor) [inline CONTRIBUTING.md:65]: That's a good question ;d Originally we had the convention as in UR: camelCase for function names and snake_case for type names. However, I think now we also have some files where functions are named using snake_case - it would be nice to unify that.

File: CONTRIBUTING.md
Comment (igchor) [inline CONTRIBUTING.md:127]: I think this is a good practice so I would leave it.

File: CONTRIBUTING.md
Comment (PatKamin) [inline CONTRIBUTING.md:11]: Using either with more than two alternatives makes me unease. Maybe rephrase it a bit:
```
Below you'll find instructions on how to contribute to UMF. This can be done in three ways:
- open a PR with code changes
- file an issue with a problem to solve
- file an issue with an enhancement proposal.
```

File: CONTRIBUTING.md
Comment (PatKamin) [inline CONTRIBUTING.md:59]: Use portable CMake commands.

File: CONTRIBUTING.md
Comment (PatKamin) [inline CONTRIBUTING.md:52]: Remember to add a link to README (or even the exact section of it).

File: CONTRIBUTING.md
Comment (PatKamin) [inline CONTRIBUTING.md:62]: (...) clang-format version - exactly (...)

File: CONTRIBUTING.md
Comment (bratpiorka) [reply CONTRIBUTING.md:65]: Please remove this section for the moment - I added a JIRA UMFW-248 for changing code to use single naming convention in all files + adding a section about it here to this doc

**Review by igchor — APPROVED** [review]

**Review by bratpiorka — APPROVED** [review]

**Comment by lukaszstolarczuk** [comment]
I've rebased, made a little update in Readme file (added a link to CONTRIBUTING.md) and updated a little content of code-style and test-and-build sections.

File: CONTRIBUTING.md
Comment (PatKamin) [inline CONTRIBUTING.md:7]: Either remove this or move to the top of this list.

File: CONTRIBUTING.md
Comment (PatKamin) [inline CONTRIBUTING.md:64]: commands -> CMake targets ?

File: CONTRIBUTING.md
Comment (PatKamin) [inline CONTRIBUTING.md:96]: I thought that signing-off wasn't meant to be a requirement in this project, @bratpiorka ?

File: CONTRIBUTING.md
Comment (lukaszstolarczuk) [reply CONTRIBUTING.md:7]: why? it doesn't break the list and when we add this section it's going to be right about here 😉

File: CONTRIBUTING.md
Comment (lukaszstolarczuk) [reply CONTRIBUTING.md:64]: Done.

File: CONTRIBUTING.md
Comment (lukaszstolarczuk) [reply CONTRIBUTING.md:96]: It's still not a requirement - it's merely an ask 😉 Also, for people in our organization this may not be needed, but perhaps for outside contributors, it's a good thing to ask...?

**Review by PatKamin — APPROVED** [review]

File: CONTRIBUTING.md
Comment (bratpiorka) [inline CONTRIBUTING.md:38]: Please add a section about the review. Add info that "codeowners" would get a notification when PR is created, but one could always add specific people if needed which should speed up the review process. Also add info that PR is merged when CI passes, all discussions are resolved and PR got at least two approves.

File: CONTRIBUTING.md
Comment (bratpiorka) [inline CONTRIBUTING.md:73]: Add that the issues in formatting will fail CI and block the merge

File: CONTRIBUTING.md
Comment (bratpiorka) [inline CONTRIBUTING.md:92]: Add that all new files should start from a license.

File: CONTRIBUTING.md
Comment (bratpiorka) [reply CONTRIBUTING.md:96]: I would remove it now. I am not sure if even my approvals are signed. This is also not tested in CI. IMO this whole file contains not questions, but requirements that would fail CI if not met.

**Review by bratpiorka — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md
Comment (bratpiorka) [reply CONTRIBUTING.md:130]: Again - could we test this? Even if not now, then in the future? If we leave it, the whole UMF team should respect it and we should not merge any non-signed-off code. But if it is a 'nice to have', I would not mention it to avoid unnecessary misunderstandings and discussions.

Comment (lukaszstolarczuk) [reply CONTRIBUTING.md:130]: heh, we had a clause like this in all(?) our previous repos, and I cannot remember a single discussion about it 😉 
We can test anything, including this (but I hope there's a tool for that 😉 ).

// examples: [PMDK](https://github.com/pmem/pmdk/blob/master/CONTRIBUTING.md) / [pmemkv](https://github.com/pmem/pmemkv/blob/master/CONTRIBUTING.md)

That said, we can use similar wording to one in PMDK, to be more flexible about it:

>In case of any doubt, the gatekeeper may ask you to certify the above in writing, i.e. via email or by including a Signed-off-by: line at the bottom of your commit comments.

Comment (bratpiorka) [reply CONTRIBUTING.md:130]: "Anecdotal evidence" :D I'm ok with "In case of any doubt" form. If we want to make it stronger we have to follow it + add tests.

File: CONTRIBUTING.md
Comment (bratpiorka) [inline CONTRIBUTING.md:87]: I'm not an expert but I think this could sound better:

To get your Pull Request (PR) to be merged, you must meet several requirements - the basic are:
- The project compiles correctly,
- ....

Comment (lukaszstolarczuk) [reply CONTRIBUTING.md:87]: This way is less fun 😄😉 Rewritten.

File: CONTRIBUTING.md
Comment (lukaszstolarczuk) [reply CONTRIBUTING.md:130]: Rewritten.

**Review by bratpiorka — APPROVED** [review]

File: CONTRIBUTING.md
Comment (PatKamin) [inline CONTRIBUTING.md:122]: such (a?) header **looks** like

File: CONTRIBUTING.md
Comment (PatKamin) [inline CONTRIBUTING.md:95]: They -> The

Comment (lukaszstolarczuk) [reply CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md
Comment (lukaszstolarczuk) [reply CONTRIBUTING.md]: Done.

**Review by PatKamin — APPROVED** [review]

**Pull request merged**
